gcp_cost_report/main.py

```diff
@@ -117,14 +119,23 @@ def try_cast_float(f):
         return None
 
 
+def month_progress() -> float:
+    """Return the percentage we are through the month"""
+    monthrange = calendar.monthrange(date.today().year, date.today().month)[1]
+    return date.today().day / monthrange
+
+
 def gcp_cost_report(unused_data, unused_context):
     """Main entry point for the Cloud Function."""
 
     totals = defaultdict(lambda: defaultdict(lambda: defaultdict(float)))
     budgets = budget_client.list_budgets(parent=f'billingAccounts/{BILLING_ACCOUNT_ID}')
     budgets_map = {b.display_name: b for b in budgets}
 
-    def format_billing_row(project_id, fields, currency):
+    # Work out what percentage of the way we are through the month
+    percent_threshold = month_progress()
+
+    def format_billing_row(project_id, fields, currency, percent_threshold=0):
```

Comment:
I know you didn't write this initially, but could we add type hints to function declarations?